### PR TITLE
feat(vowTools): add asVow helper

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
refs: #9449

## Description

- Add `asVow` helper to `VowTools` to ensure we always return vows, even in the event of early synchronous errors.
- Implement the helper in `ChainAccountKit` against #9562

### CODE REVIEW

File: packages/vow/src/vow-utils.js [hunk 1]
```diff
@@ -73,3 +74,25 @@ export const toPassableCap = k => {
...
+   * @returns {Vow<Awaited<T>>}
+   */
+  const asVow = fn => {
+    const kit = makeVowKit();
+    try {
+      kit.resolver.resolve(fn());
+    } catch (e) {
+      kit.resolver.reject(e);
+    }
+    return kit.vow;
```
Comment (mhofman) [inline packages/vow/src/vow-utils.js]: I think we could simply support not resolving a vow to a vow with the following:

```suggestion
    let result;
    try {
      result = fn();
    } catch (e) {
      result = Promise.reject(e);
    }
    if (isVow(result)) {
      return result;
    }
    const kit = makeVowKit();
    kit.resolver.resolve(result);
    return kit.vow;
```

File: packages/vow/test/asVow.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// @ts-check
+import test from 'ava';
+
+import { makeHeapZone } from '@agoric/base-zone/heap.js';
+
+import { prepareVowTools } from '../src/tools.js';
+import { isVow } from '../src/vow-utils.js';
+
+test('asVow takes a function that throws/returns synchronously and returns a vow', async t => {
```
Comment (mhofman) [inline packages/vow/test/asVow.test.js:10]: If we do the above, maybe we can add a test that verifies the pass through

**Review by mhofman — COMMENTED** [review]
Very quick review, LGTM

File: packages/vow/test/asVow.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+});
+
+test('asVow does not resolve a vow to a vow', async t => {
+  const { watch, asVow } = prepareVowTools(makeHeapZone());
+
+  const testVow = watch(Promise.resolve('payload'));
+  const testVowAsVow = asVow(() => testVow);
+
+  t.is(testVow, testVowAsVow, 'asVow does not rewrap vows');
+  t.is(await testVow, await testVowAsVow, 'result is preserved');
```
Comment (mhofman) [inline packages/vow/test/asVow.test.js]: While it's arguably overkill to check the result is preserved if the vows are the same, I assume you do want to test what it says :)
```suggestion
  t.is(await when(testVow), await when(testVowAsVow), 'result is preserved');
```

File: packages/vow/test/asVow.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  t.is(await when(watch(vowWithReturn)), 'early return');
+});
+
+test('asVow does not resolve a vow to a vow', async t => {
+  const { watch, asVow } = prepareVowTools(makeHeapZone());
+
+  const testVow = watch(Promise.resolve('payload'));
+  const testVowAsVow = asVow(() => testVow);
+
+  t.is(testVow, testVowAsVow, 'asVow does not rewrap vows');
```
Comment (mhofman) [inline packages/vow/test/asVow.test.js]: So technically we should do a "passable compare" here. The tagged vow could be a different object, but the nested payload must contain the same remotable. Not sure if we have a common way to do such passable comparisons.

**Review by mhofman — APPROVED** [review]
LGTM once the small test feedback is addressed

File: packages/vow/test/asVow.test.js (hunk 4 above)
Comment (0xpatrickdev) [reply packages/vow/test/asVow.test.js]: The closest thing I'm aware of is `getInterfaceOf`, which doesn't quite seem to fit the bill for a "passable compare". 

In this testing context, where there's no disconnection, this was a cheap way to verify the behavior. I added a comment with this feedback and context above this line.

File: packages/vow/test/asVow.test.js (hunk 3 above)
Comment (0xpatrickdev) [reply packages/vow/test/asVow.test.js]: 😅 thank you

**Pull request merged**